scvi/external/sysvi/_model.py

```diff
@@ -0,0 +1,336 @@
+from __future__ import annotations
+
+import logging
+from collections.abc import Sequence
+
+import numpy as np
+import pandas as pd
+import torch
+from anndata import AnnData
+from typing_extensions import Literal
+
+from scvi import REGISTRY_KEYS
+from scvi.data import AnnDataManager
+from scvi.data.fields import (
+    LayerField,
+    ObsmField,
+)
+from scvi.model.base import BaseModelClass, UnsupervisedTrainingMixin
+from scvi.utils import setup_anndata_dsp
+
+from ._module import SysVAE
+from ._trainingplans import TrainingPlanCustom
+from ._utils import prepare_metadata
+
+logger = logging.getLogger(__name__)
+
+
+class TrainingCustom(UnsupervisedTrainingMixin):
+    """Train method with custom TrainingPlan."""
+
+    # TODO could make custom Trainer (in a custom TrainRunner) to have in init params for early stopping
+    #  "loss" rather than "elbo" components in available param specifications - for now just use
+    #  a loss that is against the param specification
+
+    _training_plan_cls = TrainingPlanCustom
+
+
+class SysVI(TrainingCustom, BaseModelClass):
+    """Integration model based on cVAE with optional VampPrior and latent cycle-consistency loss.
+
+    Parameters
+    ----------
+    adata
+        AnnData object that has been registered via :meth:`~scvi-tools.SysVI.setup_anndata`.
+    prior
+        The prior distribution to be used. You can choose between "standard_normal" and "vamp".
+    n_prior_components
+        Number of prior components in VampPrior.
+    pseudoinputs_data_indices
+        By default VampPrior pseudoinputs are randomly selected from data.
+        Alternatively, one can specify pseudoinput indices using this parameter.
+    **model_kwargs
+        Keyword args for :class:`~scvi.external.sysvi.module.Module`
```

Comment:
```suggestion
        Keyword args for :class:`~scvi.external.sysvi.SysVAE`
```
Also need to add `SysVAE` to `__init__.py`